rest/rest.go

```diff
@@ -36,3 +39,20 @@ type Endpoint struct {
 	AllowedDuringShutdown bool // Whether we should return Unavailable Error (503) if daemon is shutting down.
 	AllowedBeforeInit     bool // Whether we should return Unavailabel Error (503) if the daemon has not been initialized (is not yet part of a cluster).
 }
+
+// Resources represents all the resources served over the same path.
+type Resources struct {
+	Path      EndpointType
+	Endpoints []Endpoint
+}
+
+// EndpointType is a type specifying the endpoint on with the resource exists.
+type EndpointType client.EndpointType
+
+// Server contains configuration and handlers for additional listeners to be instantiated after app startup.
+type Server struct {
+	Protocol    string
+	Address     types.AddrPort
```

Comment:
Question about the address, if a downstream app opt in to setup its own API server, this address would need to be a user configured setting correct?